src/optimizer/query_to_operator_transformer.cpp

```diff
@@ -50,6 +71,78 @@ void QueryToOperatorTransformer::Visit(common::ManagedPointer<parser::SelectStat
   predicates_ = {};
   transaction::TransactionContext *txn_context = accessor_->GetTxn().Get();
 
+  if (!op->GetSelectWith().empty()) {
```

Comment:
this is a pretty painful function to read, can you add some guide comments in front of each block of code?